packages/toolpad-core/src/shared/Link.tsx

```diff
@@ -5,15 +5,17 @@ import { RouterContext } from './context';
  * @ignore - internal component.
  */
 
-export interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+export interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> {
   history?: 'auto' | 'push' | 'replace';
+  href: string;
+  to: string;
 }
 
 export const Link = React.forwardRef(function Link(
   props: LinkProps,
   ref: React.ForwardedRef<HTMLAnchorElement>,
 ) {
-  const { children, href, onClick, history, ...rest } = props;
+  const { children, href, onClick, history, to, ...rest } = props;
```

Comment:
We're just passing `href` internally. Remap correctly to `to` in the `react-router` adapter,